### DISCUSSION TITLE
I need help installing avbroot on oxygenos 11 android oneplus 8 pro stock firmware

### DISCUSSION THREAD

**Discussion by floean** [chenxiaolong/avbroot#195] [Q&A]
I need help installing avbroot on oxygenos 11 android oneplus 8 pro stock firmware

**Comment by chenxiaolong** [comment]
Please post more details. What issues are you running into?

**Reply by floean** [reply]
I get bootlop on android 11 and 13.

**Reply by floean** [reply]
maybe you can help me install it? via TeamViewer

**Reply by chenxiaolong** [reply]
Sorry, I don't do live support.

Can you try flashing the patched ROM while the bootloader is unlocked and see if it boots?

**Reply by floean** [reply]
![изображение](https://github.com/chenxiaolong/avbroot/assets/67872429/d6a5c972-b99e-4751-80ca-a07d6f97d708)

**Reply by floean** [reply]
please help me

**Reply by floean** [reply]
oxygenos android 13

**Reply by chenxiaolong** [reply]
I Googled the error message and it seems like this is a common thing on the OnePlus 8 series.

I can't find any technical details of why the bootloader complains about the boot image though, so I don't really know what I can change in avbroot to work around the issue.

In the meantime, you'll probably have to go back to the stock firmware and not use avbroot.

**Reply by floean** [reply]
this does not only happen on oneplus 8 pro I think its to do with Oppo taking over OnePlus and them moving to coloros. I think they either rewrote the early stage firmwares for the 8 pro using their firmwares as a base (instead of OnePlus) oxygenos 12-13 android

How then to do it on earlier versions of android 10-11 there is no file metadata.pb what to do then in this case?

**Reply by chenxiaolong** [reply]
> How then to do it on earlier versions of android 10-11 there is no file metadata.pb what to do then in this case?

The old style OTA files are completely different--both how the partitions are stored and how the OTA zip is signed. Significant parts of avbroot would need to be rewritten to support those files unfortunately.

**Reply by floean** [reply]
is that possible?

**Reply by chenxiaolong** [reply]
> The old style OTA files are completely different

Sorry, I was wrong about this statement ^^. There's 3 OTA formats:

1. Oldest format: OTA zip containing `system.new.dat.br`/`system.new.dat` and `META-INF/com/google/android/updater-script`
2. Older format: OTA zip containing `payload.bin` and plain text `META-INF/com/android/metadata`
3. Current format: OTA zip containing `payload.bin` and protobuf `META-INF/com/android/metadata.pb`

I downloaded OxygenOS 11.0.11.11 for the OnePlus 8 Pro and it looks like it's using format (2). avbroot currently only supports format (3). Format (1) is too different and I probably will never add support for it. (2) might be possible to easily add--I'll look into it tomorrow.

**Reply by chenxiaolong** [reply]
avbroot 2.3.0 (https://github.com/chenxiaolong/avbroot/releases/tag/v2.3.0) should be able to patch Android 11 OTAs.

**Reply by floean** [reply]
how can i flash .zip.patched file in oneplus stock recavery there is no adb sideload?

**Reply by floean** [reply]
![изображение](https://github.com/chenxiaolong/avbroot/assets/67872429/cc6ba249-0443-4f46-9c0e-590cc50f9ab4)
I get these files OxygenOS 11.0.11.11 I flash them I try to lock the bootloader, but the recovery does not boot, reboots back to fastboot

**Reply by chenxiaolong** [reply]
Just to confirm: before you tried flashing the patched file, did you first flash the original zip while the bootloader was still unlocked?

I didn't document that in the README (I should), but the instructions assume that the device is already running the ROM.

**Reply by floean** [reply]
Yes, the ROM was already installed original

**Reply by chenxiaolong** [reply]
If that's the case, I'm really not sure why it wouldn't be able to boot into recovery.

Can you try one last thing: try patching the OTA with `--rootless` and remove the `--magisk <apk file>` option. This will create an unrooted OTA, but it'll help test if Magisk is potentially the issue. If that's also unable to boot into recovery, then I'm out of ideas.

**Reply by floean** [reply]
this happens when fastboot flash recovery recovery.img because of this does not start recovery

**Comment by godspeed0310** [comment]
Just to add to this, I flashed Nameless AOSP on my Oneplus 8T a couple months back and it seems to work perfectly till the time the bootloader is unlocked. So I'm guessing the issue was solved with the later versions of avbroot

**Reply by olbb** [reply]
Hi, bro. What was the Android version and OS version of your OnePlus 8T before upgrading to Nameless AOSP? I saw in #212 that OnePlus only supports avb_custom_key before Android 12. So I'm a little worried whether downgrading the system can make avb_custom_key work properly.

**Reply by godspeed0310** [reply]
Afaik I was on OOS11. Then I used wishmasterflo's firmware flasher to flash OOS13.1 firmware. I don't remember if I flashed the avb key before flashing the firmware or not though. Nameless is based on Android 14.

**Reply by olbb** [reply]
Okay, thanks. I'll try it out over the weekend
